sdk/python/lib/pulumi/invoke.py

```diff
@@ -131,10 +135,137 @@ def merge(
             if source.plugin_download_url is None
             else source.plugin_download_url
         )
-        dest.version = dest.version if source.version is None else source.version
+        dest.version = (
+            dest.version if (source.version in [None, ""]) else source.version
+        )
+
+        return dest
+
+
+class InvokeOutputOptions(InvokeOptions):
+    """
+    InvokeOutputOptions is a bag of options that control the behavior of a call to runtime.invoke_output.
+    """
+
+    depends_on: Optional["Input[Union[Sequence[Input[Resource]], Resource]]"]
+    """
+    If provided, declares that the currently-constructing resource depends on the given resources.
+    """
+
+    def __init__(
+        self,
+        parent: Optional["Resource"] = None,
+        provider: Optional["ProviderResource"] = None,
+        version: Optional[str] = "",
+        plugin_download_url: Optional[str] = None,
+        depends_on: Optional[
+            "Input[Union[Sequence[Input[Resource]], Resource]]"
+        ] = None,
+    ) -> None:
+        super().__init__(parent, provider, version, plugin_download_url)
```

Comment:
The paranoid person in me wonders if we should kwarg these for safety e.g. `__init__(parent=parent, provider=provider, ...)` etc. to avoid a potential mix-up of e.g. version and plugin download URL.